.github/workflows/ci.yaml

```
@@ -0,0 +1,70 @@
+# Copyright 2025 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI Pipeline
+
+on:
+  push:
+    branches:
+      - "pull-request/[0-9]+"
+      - main
+      - release-*
+
+jobs:
+  code-scanning:
+    uses: ./.github/workflows/code_scanning.yaml
+
+  prepare-variables:
+    needs: code-scanning
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+      build_multi_arch_images: ${{ steps.build_multi_arch_images.outputs.value }}
+    steps:
+      - name: Generate Commit Short SHA
+        id: version
+        run: echo "version=$(echo $GITHUB_SHA | cut -c1-8)" >> "$GITHUB_OUTPUT"
+
+      - name: Convert Repository Owner to Lowercase
+        id: lowercase_repo_owner
+        run: echo "lowercase_repo_owner=$(echo "${GITHUB_REPOSITORY_OWNER}" | awk '{print tolower($0)}')" >> $GITHUB_OUTPUT
+
+      - name: Determine Multi-Arch Build Setting
+        id: build_multi_arch_images
+        run: |
+          if [[ "${{ github.ref_name != 'main' && !startsWith(github.ref_name, 'release-') }}" == "true" ]]; then
+            echo "build_multi_arch_images=false" >> $GITHUB_OUTPUT
+          else
+            echo "build_multi_arch_images=true" >> $GITHUB_OUTPUT
+          fi
```

It may be clearer to invert the conditional:
```suggestion
          if [[ "${{ github.ref_name == 'main' || startsWith(github.ref_name, 'release-') }}" == "true" ]]; then
            echo "build_multi_arch_images=true" >> $GITHUB_OUTPUT
          else
            echo "build_multi_arch_images=false" >> $GITHUB_OUTPUT
          fi
```

Actually, could we simplify this further to:
```suggestion
          multi_arch_required="${{ github.ref_name == 'main' || startsWith(github.ref_name, 'release-') }}" 
          echo "build_multi_arch_images=${{ multi_arch_required }}" >> $GITHUB_OUTPUT          
```